src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCaller.java

```diff
@@ -0,0 +1,752 @@
+        if (Double.isNaN(segment.getMinorAlleleFractionPosteriorSummary().getCenter())) {
+            return CNLOHLoHCall.NO_CALL;
+        } else if (((m == getNormalNumCopies()) && (n == 0)) || ((n == getNormalNumCopies()) && (m == 0))
+            && (rho > getRhoThreshold())) {
+            return CNLOHLoHCall.CNLOH;
+        } else {
+            return CNLOHLoHCall.NOT_CNLOH;
+        }
+    }
+
+    private double[] sumOverSegs(JavaRDD<double[]> eZskBySeg) {
+        final double[][] eZskBySeg2D = eZskBySeg.collect().stream().toArray(double[][]::new); // S x K
+        final RealMatrix eZskBySegMatrix = MatrixUtils.createRealMatrix(eZskBySeg2D);
+        return GATKProtectedMathUtils.columnSums(eZskBySegMatrix);
+    }
+
+    private double[] sumOverFirstDimension(final double[][][] data) {
+        final double [] result = new double[data.length];
+        for (int i = 0; i < data.length; i++) {
+            result[i] = 0;
+            for (int j = 0; j < data[0].length; j++) {
+                for (int k = 0; k < data[0][0].length; k++) {
+                    result[i] += data[i][j][k];
+                }
+            }
+        }
+
+        return result;
+    }
+
+    @VisibleForTesting
+    static Pair<Integer, Integer> max2dIndices(final double[][] data) {
+        double maxValue = Double.NEGATIVE_INFINITY;
+        Pair<Integer, Integer> result = null;
+        for (int i=0; i < data.length; i++) {
+            for (int j=0; j < data[0].length; j++) {
+                if (data[i][j] > maxValue) {
+                    maxValue = data[i][j];
+                    result = Pair.create(i, j);
+                }
+            }
+        }
+        return result;
+    }
+
+    private int maxIndex(final double[] data) {
+        double maxValue = Double.NEGATIVE_INFINITY;
+        int result = -1;
+        for (int i=0; i < data.length; i++) {
+            if (data[i] > maxValue) {
+                maxValue = data[i];
+                result = i;
+            }
+        }
+        return result;
+    }
+
+    private double[] getFirstDimensionArray(final double[][][] array, final int dim2Index, final int dim3Index) {
+        final double[] result = new double[array.length];
+        for (int i = 0; i < result.length; i++) {
+            result[i] = array[i][dim2Index][dim3Index];
+        }
+        return result;
+    }
+
+    @VisibleForTesting
+    static double[][] sumOnlyFirstDimension(final double[][][] data) {
+        final double [][] result = new double[data[0].length][data[0][0].length];
+
+        for (int j = 0; j < data[0].length; j++) {
+            for (int k = 0; k < data[0][0].length; k++) {
+                result[j][k] = 0;
+                for (int i = 0; i < data.length; i++) {
+                    result[j][k] += data[i][j][k];
+                }
+            }
+        }
+
+        return result;
+    }
+
+    private double[] sumOverSecondDimension(final double[][][] data) {
```

Comment:
I agree that "sum over" here and above is weird.  Given that these are private methods, I think that names like "sumOverIndicesIJ" or even just "sumOverIJ" might be clear enough ("sumOverFirstAndSecondDimensions" seems a bit verbose to me).  But I'll leave it up to the two of you!